core/dbt/events/types.py

```diff
@@ -2170,6 +2190,9 @@ def message(self) -> str:
 @dataclass
 class NodeFinished(DebugLevel, Cli, File):
     unique_id: str
+    report_node_data: ParsedModelNode
+    node_status: str = node_statuses['pass']
+    node_state: str = node_states['success']
```

Comment:
I wonder if it makes sense to include the full [contracted `RunResult`](https://github.com/dbt-labs/dbt-core/blob/7f2d3cd24f8715d3e36873a1f0049d8a764b2df5/core/dbt/contracts/results.py#L117-L127) object in `NodeFinished` as well — even if we don't use it to template out the CLI message or specific `node_info` — that way, users can access its serialized representation from `data` within structured logs